keanu-project/src/main/java/io/improbable/keanu/plating/Plates.java

```diff
@@ -31,4 +31,12 @@ public void add(Plate p) {
     public Iterator<Plate> iterator() {
         return containedPlates.iterator();
     }
+
+    public Plate getLastPlate() {
+        try {
+            return this.asList().get(this.size() - 1);
+        } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
That's interesting, I always felt that this way was more elegant. Any particular reason why you favour that way?

I'm just asking out of interest, I'm happy to change it.

Reply:
IMHO exceptions like ArrayIndexOutOfBoundsException and NullPointerException aren't meant to be caught unless you don't have a choice. It probably comes from the older days where you'd get a seg fault or something really nasty if you didn't check the bounds before accessing. I don't think the code should rely (unless there isn't a choice) on being told that there was an exception if the code can be written in a way that is safe to begin with. This might be handled this way in Python or another language but I'd prefer to avoid it here.

Reply:
Thanks, that's a good perspective. I'll look out for it in other parts of my code.